charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -42,7 +42,8 @@ Build config file for daemonset OpenTelemetry Collector
 {{- $data := dict "Values" $values | mustMergeOverwrite (deepCopy .) }}
 {{- $config := include "opentelemetry-collector.baseConfig" $data | fromYaml }}
 {{- $config := include "opentelemetry-collector.ballastConfig" $data | fromYaml | mustMergeOverwrite $config }}
-{{- $config := mustMergeOverwrite (include "opentelemetry-collector.daemonset.containerLogsConfig" $data | fromYaml) $config }}
+{{- $config := mustMergeOverwrite (include "opentelemetry-collector.daemonset.logsCollectionConfig" $data | fromYaml) $config }}
+{{- $config := mustMergeOverwrite (include "opentelemetry-collector.daemonset.hostMetricsConfig" $data | fromYaml) $config }}
```

Comment:
```
{{- if .Values.presets.hostMetrics.enabled }}
```
should probably go here. 

What if instead of `mustMergeOverwrite` we have `include "opentelemetry-collector.daemonset.applyHostMetricsConfig" $config ` that would do list merges where needed, e.g. `.service.pipelines.metrics.receivers | append "hostmetrics" | uniq`?

Reply:
I think we still have to do a `mustMergeOverwrite` for the actual component configuration, but I like the `append` idea for dealing with the pipelines.  I'll look into that tomorrow.

Reply:
> I think we still have to do a mustMergeOverwrite for the actual component configuration

Yes, but inside `opentelemetry-collector.applyHostMetricsConfig` for particular parts of the config

Reply:
@dmitryax there is still cleanup work today but I think I got the "append" solution working